internal/jujucommands/jujucommands.go

```diff
@@ -5,127 +5,86 @@
 package jujucommands
 
 import (
+	"context"
 	"fmt"
-	"io"
-	"strings"
+	"os/exec"
+	"sync"
 
-	"github.com/juju/cmd/v3"
-	"github.com/juju/juju/cmd/juju/commands"
-	"github.com/juju/juju/jujuclient"
 	"github.com/mitchellh/go-linereader"
-)
-
-var (
-	whitelist = []string{"help", "bootstrap"}
+	"mvdan.cc/sh/v3/shell"
 )
 
 type outputLine struct {
 	Line string
 	Err  error
 }
 
-// runCmdWithOutputRetriever runs the command given by cmdAndArgs using the provided jujuclient.ClientStore.
-// It returns a channel that streams OutputLine values for each line of command output and any final error.
-//
-// Upon receiving an error in an OutputLine, the line previously streamed is guaranteed to be the error line written by
-// the command.
-// I.e., if you run: "help -b"
-// You will get:
-//
-//	Output line: ERROR option provided but not defined: -b // The error line from juju (Which is a none-erroneous OutputLine)
-//	Command finished with error: cmd failed with code 2 // The command error code line (Contains an error in the OutputLine)
-//
-// Consumers are expected to simply read from the OutputLine channel and check for an error like so:
-//
-//	for out := range outputCh {
-//		if out.Err != nil {
-//			fmt.Println("Command finished with error:", out.Err)
-//			break
-//		}
-//		fmt.Println("Output line:", out.Line)
-//	}
-//
-// The OutputLine channel is closed once all output from the command has finished an the error has been captured.
-//
-// Lines returned are returned with no newlines.
-func runCmdWithOutputRetriever(store jujuclient.ClientStore, cmdAndArgs string) (<-chan outputLine, error) {
-	cmdReader, cmdWriter := io.Pipe()
-
-	cmdCtx, err := cmd.DefaultContext()
+// runJujuCmd runs a juju command with the given command string and JUJU_DATA directory.
+// It returns a channel that will receive output lines from the command's stdout and stderr.
+// The command is run in a separate goroutine, and the context can be used to cancel the command.
+func runJujuCmd(ctx context.Context, cmdStr, jujuDataDir string) (<-chan outputLine, error) {
+	env := func(key string) string {
+		if key == "JUJU_DATA" {
+			return jujuDataDir
+		}
+		return ""
+	}
+	args, err := shell.Fields(cmdStr, env)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse command: %w", err)
+	}
+
+	cmd := exec.Command("juju", args...)
```

Comment:
The env vars are being converted to args here? I would've expected to see `cmd.Env()` being used.